src/acl/Acl.h

```diff
@@ -30,8 +31,26 @@ typedef ACL *(*Maker)(TypeName typeName);
 /// use the given ACL Maker for all ACLs of the named type
 void RegisterMaker(TypeName typeName, Maker maker);
 
+class ProxyAuthMatchCacheEntry
+{
+public:
+    ProxyAuthMatchCacheEntry(int matchRv, void *aclData) :
+        matchrv(matchRv),
+        acl_data(aclData)
+    {
+    }
+
+    int matchrv;
+    void *acl_data;
+};
+
+/// LRU cache used for User ACL match
```

Comment:
The LRU addition is useful, thank you. If you can also describe this type without relying on how it is currently used, please do so (otherwise we get these recursive/reentrant descriptions that essentially document nothing). For example, I cannot tell what "User ACL" is in this context even though I do know that there is a User class/concept and and ACL class/concept.

```suggestion
/// an LRU-driven cache of past proxy_auth matching results; lookups use linear search
```

Please note that the above suggestion may be factually wrong! I do not know exactly what this type is. I am just trying illustrate that you can define this type without referring to classes that use it.

Reply:
BTW, the "past proxy_auth matching result" part in my suggestion (or its corrected replacement) really belongs to the new ProxyAuthMatchCacheEntry class description (which is currently undocumented).